test/e2e/helm.go

```diff
@@ -0,0 +1,72 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"os/exec"
+
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/test/framework"
+)
+
+// HelmOptions handles arguments to a `helm install` command.
+type HelmOptions struct {
+	StringValues []string // --set-string
+	ValueFiles   []string // --values
+	Values       []string // --set
+}
+
+// InstallHelmChart takes a Helm repo URL, a chart name, and release name, and creates a Helm release on the E2E workload cluster.
+func InstallHelmChart(ctx context.Context, clusterProxy framework.ClusterProxy, namespace, repoURL, chartName, releaseName string, options *HelmOptions, version string) {
```

Comment:
The previous implementation has a lot of built-in resiliency with retries and checking if the release already exists before installing which was added over time to address test flakes. Is `helm install` idempotent and somewhat equivalent? Are there any low-hanging fruits to make this new implementation closer to that (eg. skip install if the release already exists, uninstall the release if the install fails, etc.)

A lot of this should be temporary as it will soon be replaced by #3736 so we shouldn't put too much time/effort into this "quick fix" but if there's anything we can do in the meantime to make it not regress stability that might save us some retests.

Reply:
Good thinking: `helm upgrade --install` is the idempotent way to install a chart apparently, and `--atomic` should clean things up in case of failure. I'll make those changes.

We're already retrying here the same as before this change. 

The only real difference remaining is that we're not checking to see if the release exists before installing it, but I don't think that's needed.

Reply:
> The only real difference remaining is that we're not checking to see if the release exists before installing it, but I don't think that's needed.

I think that is needed in cluster upgrade scenarios where the addon is already installed on the cluster, but `helm upgrade` should cover it?

Reply:
> `helm upgrade` should cover it?

I think so? CI will tell the tale.

Reply:
I agree, I think it makes to not polish this a ton since it's going to be replaced soon.